File: .yalc/@entur/atb-mobile-client-sdk/token/token-core-android-lib/src/main/java/no/entur/abt/android/token/attestation/DeviceAttestator.java
package no.entur.abt.android.token.attestation;

import com.google.protobuf.GeneratedMessageLite;

/**
 * Create device attestations.
 */
public interface DeviceAttestator {

	/**
	 * Create attestation of device with serverGeneratedNonce from server.
	 * 
	 * @param serverGeneratedNonce a short lived secret from the server
	 * @param signaturePublicKey   the public signature key to attest
	 * @param encryptionPublicKey  the public encryption key to attest
	 * @return
	 */
	GeneratedMessageLite attest(byte[] serverGeneratedNonce, byte[] signaturePublicKey, byte[] encryptionPublicKey) throws DeviceAttestationException;

	/**
	 * Attestation check
	 *
	 * @return true if (real) device attestation works. So typically false for an emulator.
	 */

	boolean supportsAttestation();

}

File: .yalc/@entur/atb-mobile-client-sdk/token/token-core-ios-lib/AbtTokenCoreSDK/attestation/DeviceAttestator.swift
import Foundation

public protocol DeviceAttestator {
    /// Create attestation of device with serverGeneratedNonce from server.
    ///
    /// `serverGeneratedNonce` A short lived secret from the server
    /// `signatureKeyPair` The public signature key to attest
    /// `encryptionKeyPair` The public encryption key to attest
    /// `attestationEncryptionPublicKey`
    /// `onComplete` A callback to return `AttestionObject` object
    func attest(serverGeneratedNonce: Data, signatureKeyId: String?, signatureKeyPair: KeyPair?, encryptionKeyPair: KeyPair, attestationEncryptionPublicKey: Data, onComplete callback: @escaping (Result<AttestionObject, DeviceAttestatorError>) -> Void)

    /// `reattestationData` A base64 encoded string with `AttestationData` object
    func reAttest(reattestationData: String, signatureKeyId: String?, signatureKeyPair: KeyPair?, encryptionKeyPair: KeyPair, onComplete callback: @escaping (Result<AttestionObject, DeviceAttestatorError>) -> Void)

    func supportsAttestation() -> Bool
}
